manifest.go

```diff
@@ -23,143 +24,170 @@ type Manifest struct {
 }
 
 type rawManifest struct {
-	Dependencies map[string]possibleProps `json:"dependencies,omitempty"`
-	Overrides    map[string]possibleProps `json:"overrides,omitempty"`
-	Ignores      []string                 `json:"ignores,omitempty"`
-	Required     []string                 `json:"required,omitempty"`
+	Dependencies []rawProject `toml:"dependencies,omitempty"`
+	Overrides    []rawProject `toml:"overrides,omitempty"`
+	Ignores      []string     `toml:"ignores,omitempty"`
+	Required     []string     `toml:"required,omitempty"`
 }
 
-type possibleProps struct {
-	Branch   string `json:"branch,omitempty"`
-	Revision string `json:"revision,omitempty"`
-	Version  string `json:"version,omitempty"`
-	Source   string `json:"source,omitempty"`
+type rawProject struct {
+	Name     string `toml:"name"`
+	Branch   string `toml:"branch,omitempty"`
+	Revision string `toml:"revision,omitempty"`
+	Version  string `toml:"version,omitempty"`
+	Source   string `toml:"source,omitempty"`
 }
 
-func newRawManifest() rawManifest {
-	return rawManifest{
-		Dependencies: make(map[string]possibleProps),
-		Overrides:    make(map[string]possibleProps),
-		Ignores:      make([]string, 0),
-		Required:     make([]string, 0),
+func readManifest(r io.Reader) (*Manifest, error) {
+	buf := &bytes.Buffer{}
+	_, err := buf.ReadFrom(r)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to read byte stream")
 	}
-}
 
-func readManifest(r io.Reader) (*Manifest, error) {
-	rm := rawManifest{}
-	err := json.NewDecoder(r).Decode(&rm)
+	raw := rawManifest{}
+	err = toml.Unmarshal(buf.Bytes(), &raw)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "Unable to parse the manifest as TOML")
 	}
+
+	return fromRawManifest(raw)
+}
+
+func fromRawManifest(raw rawManifest) (*Manifest, error) {
 	m := &Manifest{
-		Dependencies: make(gps.ProjectConstraints, len(rm.Dependencies)),
-		Ovr:          make(gps.ProjectConstraints, len(rm.Overrides)),
-		Ignores:      rm.Ignores,
-		Required:     rm.Required,
+		Dependencies: make(gps.ProjectConstraints, len(raw.Dependencies)),
+		Ovr:          make(gps.ProjectConstraints, len(raw.Overrides)),
+		Ignores:      raw.Ignores,
+		Required:     raw.Required,
 	}
 
-	for n, pp := range rm.Dependencies {
-		m.Dependencies[gps.ProjectRoot(n)], err = toProps(n, pp)
+	for i := 0; i < len(raw.Dependencies); i++ {
+		name, prj, err := toProject(raw.Dependencies[i])
```

Comment:
The cost of moving to an array from a map is that it's now possible for users to express the same project more than once. This loop has to validate that each entry in the input is unique, and fail hard otherwise.